2025-03-04 12:13:20,273 ERROR: Exception on /user/susan [GET] [in C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py:875]
Traceback (most recent call last):
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\PycharmProjects\flaskProject\app\routes.py", line 80, in user
    return render_template('user.html', user=user, posts=posts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Mary\PycharmProjects\flaskProject\templates\user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\Mary\PycharmProjects\flaskProject\templates\base.html", line 30, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\PycharmProjects\flaskProject\templates\user.html", line 16, in block 'content'
    {{ form.hidden_tag() }}
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\jinja2\environment.py", line 490, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'form' is undefined
2025-03-04 12:14:13,292 ERROR: Exception on /user/susan [GET] [in C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py:875]
Traceback (most recent call last):
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\PycharmProjects\flaskProject\app\routes.py", line 80, in user
    return render_template('user.html', user=user, posts=posts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
         ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\jinja2\environment.py", line 1295, in render
    self.environment.handle_exception()
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\jinja2\environment.py", line 942, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\Mary\PycharmProjects\flaskProject\templates\user.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\Mary\PycharmProjects\flaskProject\templates\base.html", line 30, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Mary\PycharmProjects\flaskProject\templates\user.html", line 16, in block 'content'
    {{ form.hidden_tag() }}
  File "C:\Users\Mary\AppData\Local\pypoetry\Cache\virtualenvs\flaskproject-m2ZFjxRD-py3.12\Lib\site-packages\jinja2\environment.py", line 490, in getattr
    return getattr(obj, attribute)
           ^^^^^^^^^^^^^^^^^^^^^^^
jinja2.exceptions.UndefinedError: 'form' is undefined
